ppx/reason_react_ppx.ml

```diff
@@ -119,7 +142,8 @@ let transformChildrenIfListUpper ~ctxt ~loc ~mapper theList =
            Some { pexp_desc = Pexp_tuple [ v; acc ]; _ } );
      _;
     } ->
-        transformChildren_ acc (mapper#expression ctxt v :: accum)
+        transformChildren_ acc
+          (mapper#expression ctxt (wrap_constant v) :: accum)
```

Comment:
Below are mostly the points that Ricky already shared in https://github.com/reasonml/reason/pull/1910, but I think it's good to have them here for completion.

---

This introduces a new "region" where things behave differently. E.g. this compiles:

```reason
module Foo = {
  [@react.component]
  let make = () => {
    <div> "a" </div>;
  };
};
```

but this does not:

```reason
module Foo = {
  [@react.component]
  let make = () => {
    let a = "a";
    <div> a </div>;
  };
};
```

I am not sure how I feel about these "regions", it reminds me a bit of `useEffect` and the dependencies, which is probably no bueno 😅